src/cpu/rnn/rnn_reorders.hpp

```diff
@@ -858,7 +859,10 @@ struct rnn_brgemm_weights_reorder_s8_t : public primitive_t {
         const auto &blocked_d = dst_d;
         const auto &pdims = blocked_d.padded_dims();
 
-        const int o_block = pd()->otag_ == ldgOI64o4i ? 64 : 32;
+        const auto otag = pd()->otag_;
+        const int o_block = otag == ldgOI64o4i               ? 64
+                : utils::one_of(otag, ldgOI32o4i, ldOI32o4i) ? 32
+                                                             : 16;
```

Comment:
(nit) why not read inner_blk directly instead?

Reply:
Makes sense, changed it.